certgen/certProcessor/src/main/java/org/incredible/certProcessor/CertificateFactory.java

```diff
@@ -115,38 +118,55 @@ public Properties readPropertiesFile() {
         return properties;
     }
 
-    public static boolean verifySignature(String certificate, String signatureValue) {
+
+    /**
+     * to verify signature value
+     *
+     * @param certificate
+     * @param signatureValue
+     * @param properties
+     * @return
+     */
+    public boolean verifySignature(CertificateExtension certificate, String signatureValue, Map<String, String> properties) {
         boolean isValid = false;
+        SignatureHelper signatureHelper = new SignatureHelper(properties);
         try {
-            isValid = signatureHelper.verify(certificate.getBytes(),
-                    signatureValue);
-        } catch (InvalidKeyException e) {
-            e.printStackTrace();
-        } catch (SignatureException e) {
-            e.printStackTrace();
+            Map signReq = new HashMap<String, Object>();
+            signReq.put("claim", certificate);
+            signReq.put("signatureValue", signatureValue);
+            signReq.put("keyId", 2);
```

Comment:
this keyId must be fetched from the creator - certificate.getSignature().getCreator() and extract the key id.